tests/Feature/PushToGitHubTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Actions\PushToGitHub;
+use Tests\Feature\Fakes\FakeProcess;
+use Tests\TestCase;
+
+/**
+ * @group git-and-github
+ */
+class PushToGitHubTest extends TestCase
+{
+    /** @test */
+    function it_pushes_to_github()
+    {
+        config(['lambo.store.push_to_github' => true]);
+        config(['lambo.store.branch' => 'branch']);
+
+        $this->shell->shouldReceive('execInProject')
+            ->with("git -c credential.helper= -c credential.helper='!gh auth git-credential' push -u origin branch")
+            ->once()
+            ->andReturn(FakeProcess::success());
+
+        app(PushToGitHub::class)();
+    }
+
+    /** @test */
+    function it_skips_pushing_to_github()
```

Comment:
What's actually doing the testing here?

Reply:
The test is checking that `Shell->execInProject(...)` is not called—it fails with a mockery exception if it is—however, this behaviour is not apparent.

I have updated this with an implementation using a spy that explicitly checks that `execInProject(...)` is not called.